### PR TITLE
fix(dtfs2 6899): fix blocked user logging in with most recent sign in link

### PR DESCRIPTION
**Pull request by AlexBramhill**
## Introduction :pencil2:
This change fixes an edge case where a user can access a recently issued sign in link, despite being blocked.

## Resolution :heavy_check_mark:
When a user is blocked, we wipe any past tokens from the database.
We also check the user is not blocked on login. While we realistically never get to this stage, due to the method being public, it has been included to avoid future issues

## Miscellaneous :heavy_plus_sign:
Moved some legacy hand written `user-status`s to enums

### CODE REVIEW

File: portal-api/api-tests/v1/users/login-with-sign-in-link.api-test.js [hunk 1]
```diff
@@ -111,26 +121,30 @@ describe('POST /users/:userId/sign-in-link/:signInToken/login', () => {
...
             { _id: { $eq: testUser._id } },
             {
               $set: {
                 signInToken: {
                   saltHex: saltHexForValidSignInToken,
                   hashHex: nonMatchingHashHex,
-                  expiry: Date.now() + (30 * 60 * 1000),
-                }
-              }
+                  expiry: Date.now() + 30 * 60 * 1000,
```
Comment (abhi-markan) [inline portal-api/api-tests/v1/users/login-with-sign-in-link.api-test.js:145]: Can we add a comment stating `30 minutes expiry` please.

File: portal-api/api-tests/v1/users/post-sign-in-link.api-test.js [hunk 2]
```diff
@@ -130,6 +137,13 @@ describe('POST /users/me/sign-in-link', () => {
       const userInDb = await databaseHelper.getUserById(partiallyLoggedInUserId);
       expect(userInDb.signInLinkSendCount).toBe(4);
     });
+
+    it('deletes any existing sign in token data', async () => {
+      await sendSignInLink();
+
+      const userInDb = await databaseHelper.getUserById(partiallyLoggedInUserId);
+      expect(userInDb.signInToken).toBeUndefined();
```
Comment (abhi-markan) [inline portal-api/api-tests/v1/users/post-sign-in-link.api-test.js:145]: Can we also please add as another conditional check.

```javascript
expect(userInDb.signInLinkSendCount).toBe(3);
```

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Good work, minor comments to address please.

File: portal-api/api-tests/v1/users/post-sign-in-link.api-test.js (hunk 2 above)
Comment (AlexBramhill) [reply portal-api/api-tests/v1/users/post-sign-in-link.api-test.js:145]: We've got a test that does this above - `increments the signInLinkSendCount`

File: portal-api/api-tests/v1/users/login-with-sign-in-link.api-test.js (hunk 1 above)
Comment (AlexBramhill) [reply portal-api/api-tests/v1/users/login-with-sign-in-link.api-test.js:145]: added as a named variable instead of comment

**Review by abhi-markan — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**